### PR TITLE
Add support for market/location page views - for Blog #79

### PR DESCRIPTION
**Pull request by pdbreen**
* [x] requires updated drewroberts/blog prior to merge

---

Adds layouts and views for market and location pages and sets default layout when creating market / location page association.  Updates location middleware for location aware pages to automatically redirect to desired url when inbound url doesn't match.  Adds `Route::postLocation(..)` macro to register location aware post routes.

### CODE REVIEW

File: src/LocationsServiceProvider.php [hunk 1]
```diff
@@ -84,6 +84,24 @@ public function bootingPackage()
...
+                    Route::middleware(ResolveLocation::class)
+                        ->post('company/' . $uri, $action);
+
+                    Route::middleware(ResolveLocation::class)
+                        ->post('{market}/{location}/' . $uri, $action);
+
+                    Route::middleware(ResolveLocation::class)
+                        ->post('{market}/' . $uri, $action);
+                });
+        });
```
Comment (drewroberts) [inline src/LocationsServiceProvider.php]: This logic is great!

File: tests/Unit/Http/Controllers/PageControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        $location = Location::factory()->create();
+        $market = $location->market;
+
+        $this->get($this->webUrl("{$market->slug}"))
+            ->assertRedirect('/');
+
+        $this->get($this->webUrl("{$market->slug}/{$location->slug}"))
+            ->assertRedirect('/');
+    }
```
Comment (drewroberts) [inline tests/Unit/Http/Controllers/PageControllerTest.php:30]: Perfect. This is exactly how it should function with only one market & one location.

File: tests/Unit/Http/Controllers/PageControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        $location = Location::query()->first();
+        $market = $location->market;
+
+        $this->get($this->webUrl("{$market->slug}"))
+            ->assertOk()
+            ->assertSee("-- M:{$market->id} --");
+
+        $this->get($this->webUrl("{$market->slug}/{$location->slug}"))
+            ->assertRedirect("{$market->slug}");
+    }
```
Comment (drewroberts) [inline tests/Unit/Http/Controllers/PageControllerTest.php:51]: Perfect.

File: tests/Unit/Http/Controllers/PageControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        $this->get($this->webUrl("{$market->slug}"))
+            ->assertOk()
+            ->assertSee("-- M:{$market->id} --");
+
+        $this->logToStderr();
+        $this->get($this->webUrl("{$market->slug}/{$location->slug}"))
+            ->assertOk()
+            ->assertSee("-- M:{$market->id} L:{$location->id} --");
+    }
```
Comment (drewroberts) [inline tests/Unit/Http/Controllers/PageControllerTest.php]: Perfect.

**Review by drewroberts — APPROVED** [review]
Thank you! 🏦

File: resources/views/page/location/base.blade.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
+@extends('support::base')
+
+@section('content')
+    {{-- DO NOT REMOVE - page identity tag --}}
```
Comment (jasonmccreary) [inline resources/views/page/location/base.blade.php:4]: Unrelated to this PR, but you could make this a partial which may better communicate the intent and avoid the comment.

**Review by jasonmccreary — COMMENTED** [review]
Looks good. Tangential comment.

File: resources/views/page/location/base.blade.php (hunk 5 above)
Comment (drewroberts) [reply resources/views/page/location/base.blade.php:4]: CC: @chx2 

Chris, could you make this partial?

Comment (pdbreen) [reply resources/views/page/location/base.blade.php:4]: @drewroberts  The change to partials is already complete.  Ex - https://github.com/tipoff/locations/blob/faa8a3e67c6249629a34accc2b4035424c09bb39/resources/views/page/location/base.blade.php#L4

**Pull request merged**